app/scripts/components/analysis/define/index.tsx

```diff
@@ -185,12 +191,12 @@ export default function Analysis() {
     if (!readyToLoadDatasets) {
       return 'To select datasets, please define an area and a date first.';
     } else {
-      if (!selectableDatasetLayers.length) {
-        return 'No datasets available in that thematic area for the currently selected dates and area.';
-      } else if (stacSearchStatus === S_LOADING) {
+      if (stacSearchStatus === S_LOADING) {
         return 'Loading...';
       } else if (stacSearchStatus === S_FAILED) {
         return 'Error loading datasets.';
+      } else if (!selectableDatasetLayers.length) {
+        return 'No datasets available in that thematic area for the currently selected dates and area.';
```

Comment:
The order of the if statements was causing the Loading message not to be displayed